src/services/blockchain/ethereum/ethereum-blockchain-service.ts

```diff
@@ -115,20 +126,50 @@ function handleTimeoutError(transactionTimeoutSecs: number): void {
 function make(config: Config): EthereumBlockchainService {
   const ethereum = config.blockchain.connectors.ethereum
   const { host, port, url } = ethereum.rpc
+  let network
+  try {
+    network = Network.from(ethereum.network)
+  } catch (e) {
+    logger.warn(`Network ${ethereum.network} is unknown, cannot use static network`)
+  }
 
   let provider
   if (url) {
     logger.imp(`Connecting ethereum provider to url: ${url}`)
-    provider = new ethers.providers.StaticJsonRpcProvider(url)
+    if (network) {
+      provider = new ethers.JsonRpcProvider(url, network, {
+        staticNetwork: network,
+      })
+    } else {
+      provider = new ethers.JsonRpcProvider(url)
+    }
   } else if (host && port) {
     logger.imp(`Connecting ethereum provider to host: ${host} and port ${port}`)
-    provider = new ethers.providers.StaticJsonRpcProvider(`${host}:${port}`)
+    const hostPort = `${host}:${port}`
+    if (network) {
```

Comment:
maybe we can reduce code duplication by creating an empty `options` object and adding the `staticNetwork` field to it if `network` is set.  Then we can always just pass the options object to the `JSONRpcProvider` constructor and not need the extra `if(network)` conditional in both blocks?

Reply:
Will reduce the duplication.